src/providers/r2Provider.ts

```diff
@@ -98,10 +98,19 @@ function r2MetadataToHeaders(
 ): HttpResponseHeaders {
   const { httpMetadata } = object;
 
+  let contentType: string;
```

Comment:
Let's make a JSON file that can be used as a Content Type override map. Or even  constant.

So for example: { "tab": "text/plain" }

Reply:
Removes the need of specific if statements, simply attempt to retrieve value from map given the file extension, if undefined use default otherwise use override.